swaybar/ipc.c

```diff
@@ -19,13 +20,15 @@ void ipc_send_workspace_command(const char *workspace_name) {
 
 static void ipc_parse_config(struct config *config, const char *payload) {
 	json_object *bar_config = json_tokener_parse(payload);
-	json_object *tray_output, *mode, *hidden_bar, *position, *status_command;
+	json_object *tray_output, *display_mode, *hidden_state, *position, *status_command;
 	json_object *font, *bar_height, *wrap_scroll, *workspace_buttons, *strip_workspace_numbers;
 	json_object *binding_mode_indicator, *verbose, *colors, *sep_symbol, *outputs;
 	json_object *markup;
 	json_object_object_get_ex(bar_config, "tray_output", &tray_output);
-	json_object_object_get_ex(bar_config, "mode", &mode);
-	json_object_object_get_ex(bar_config, "hidden_bar", &hidden_bar);
+	// "mode" as in the bar's display mode and "mode" as in binding mode
+	// should be distinguished more carefully
+	json_object_object_get_ex(bar_config, "mode", &display_mode);
+	json_object_object_get_ex(bar_config, "hidden_state", &hidden_state);
```

Comment:
I'll yield to compatibility concerns.

... but it's only compatiblity with i3 being stupid here. Setting `mode=hide` + `hidden_state=show` appears to effectively disabling hiding altogether. Probably because (this PR does this too) the i3 event handler doesn't *really* invert the state in response to a toggle event but sets the state to hard-coded on/off values. The result is that `hidden_state=show` sets it to `on` when in the released state, and then pressed event switches it from `on->on`.

There's no reason for anyone to ever use this option in their config. It's there because remote control requires it to be an IPC command (not a config option) and my guess is i3 doesn't distinguish the two because it treats config as scripted IPC (which has its benefits). Strictly speaking, it should distinguish the two cases.

There **could** be a legitimate use case for this as a selector between modifier-hides-bar and modifier-unhides-bar behavior. Not something I Imagine ever using, and i3 doesn't use it for that.
Instead, it's more like an override for hide mode (hence bindable).

https://i3wm.org/docs/userguide.html, section 5.3
```
In order to control whether i3bar is hidden or shown in hide mode, 
there exists the hidden_state option, which has no effect in dock 
mode or invisible mode. It indicates the current hidden_state of the bar:

1. The bar acts like in normal hide mode, it is hidden and is only unhidden 
   in case of urgency hints or by pressing the modifier key (hide state), or 
2. it is drawn on top of the currently visible workspace (show state).

Like the mode, the hidden_state can also be controlled through i3, this can be 
done by using the bar hidden_state command.
```